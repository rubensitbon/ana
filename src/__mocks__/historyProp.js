const history = {
  length: 1,
  action: '',
  location: {
    pathname: '',
    search: '',
    hash: '',
    state: '',
  },
  push: jest.fn(),
  replace: jest.fn(),
  go: jest.fn(),
  goBack: jest.fn(),
  goForward: jest.fn(),
  block: jest.fn(),
};

export default history;
