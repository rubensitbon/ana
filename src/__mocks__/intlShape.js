const intlShape = {
  formatDate: jest.fn(),
  formatTime: jest.fn(),
  formatRelative: jest.fn(),
  formatNumber: jest.fn(),
  formatPlural: jest.fn(),
  formatMessage: jest.fn(),
  formatHTMLMessage: jest.fn(),
};

export default intlShape;
