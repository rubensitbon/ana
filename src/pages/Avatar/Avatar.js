// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import type { IntlShape } from 'react-intl';
import type { RouterHistory } from 'react-router-dom';

import StyledAvatar from './Avatar.style';

type Props = {
  intl: IntlShape,
  history: RouterHistory,
  fetchUser: (username: string) => void,
  updateUsername: (value: string) => void,
  userAvatarUrl: string,
  username: string,
};

class Avatar extends React.Component<Props> {
  onInputChange = (event: SyntheticInputEvent<HTMLInputElement>): void => {
    this.props.updateUsername(event.target.value);
  };

  fetchUser = (): void => {
    this.props.fetchUser(this.props.username);
  };

  navigateTo = (path: string): (() => void) => (): void => {
    this.props.history.push(path);
  };

  render() {
    const { formatMessage } = this.props.intl;
    const { userAvatarUrl } = this.props;

    return (
      <StyledAvatar>
        <button tabIndex={0} onClick={this.navigateTo('/')}>
          <FormattedMessage id="page.back" />
        </button>
        <p>{formatMessage({ id: 'page.api-to-translate-example' })}</p>
        <p>
          <input
            className="github-avatar-input"
            type="text"
            onChange={this.onInputChange}
            placeholder={formatMessage({ id: 'page.add-github-username' })}
          />
        </p>
        <p>
          <button onClick={this.fetchUser}>
            <FormattedMessage id="page.fetch-github-avatar" />
          </button>
        </p>
        {userAvatarUrl && <img src={userAvatarUrl} alt="user avatar" />}
      </StyledAvatar>
    );
  }
}

export default Avatar;
