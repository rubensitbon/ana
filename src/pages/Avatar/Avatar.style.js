// @flow
import styled from 'styled-components';

const StyledAvatar = styled.div`
  .github-avatar-input {
    width: 200px;
  }
`;

export default StyledAvatar;
