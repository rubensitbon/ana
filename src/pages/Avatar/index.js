// @flow
export { default } from './Avatar.wrap';
