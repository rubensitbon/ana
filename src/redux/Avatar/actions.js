// @flow
import {
  UPDATE_USERNAME,
  USER_FETCH_REQUEST,
  USER_FETCH_SUCCESS,
  USER_FETCH_ERROR,
} from './constant';

export function updateUsername(username: string): UpdateUsernameAction {
  return {
    type: UPDATE_USERNAME,
    payload: { username },
  };
}

export function fetchUserRequest(username: string): FetchUserRequestAction {
  return {
    type: USER_FETCH_REQUEST,
    payload: { username },
  };
}

export function fetchUserSuccess(user: User): FetchUserSuccessAction {
  return {
    type: USER_FETCH_SUCCESS,
    payload: { user },
  };
}

export function fetchUserError(error: ErrorType): FetchUserErrorAction {
  return {
    type: USER_FETCH_ERROR,
    payload: { error: error.message },
  };
}

export default {
  updateUsername,
  fetchUserRequest,
  fetchUserSuccess,
  fetchUserError,
};
