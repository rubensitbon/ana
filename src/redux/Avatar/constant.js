// @flow
export const USER_FETCH_REQUEST = 'Avatar/USER_FETCH_REQUEST';
export const USER_FETCH_SUCCESS = 'Avatar/USER_FETCH_SUCCESS';
export const USER_FETCH_ERROR = 'Avatar/USER_FETCH_ERROR';
export const UPDATE_USERNAME = 'Avatar/UPDATE_USERNAME';
