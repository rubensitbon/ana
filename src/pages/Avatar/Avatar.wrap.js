// @flow
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import type { Dispatch } from 'redux';

import { actions } from 'redux/Avatar';
import Avatar from './Avatar';

const mapStateToProps = (state: Store): AvatarState => ({
  username: state.avatar.username,
  userAvatarUrl: state.avatar.userAvatarUrl,
});

const mapDispatchToProps = (dispatch: Dispatch<*>): Object => ({
  fetchUser(username) {
    dispatch(actions.fetchUserRequest(username));
  },
  updateUsername(username) {
    dispatch(actions.updateUsername(username));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(Avatar));
