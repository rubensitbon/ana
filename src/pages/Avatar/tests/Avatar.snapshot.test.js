// @flow
import React from 'react';

import historyProp from '__mocks__/historyProp';
import intlShape from '__mocks__/intlShape';
import createComponentWithIntl from 'services/i18n/create-component-with-intl';
import Avatar from '../Avatar';

describe('The page component', () => {
  it('should render correctly', () => {
    const props = {
      intl: intlShape,
      history: historyProp,
      fetchUser: jest.fn(),
      updateUsername: jest.fn(),
      userAvatarUrl: 'https://avatars1.githubusercontent.com/u/9215726?v=4',
      username: 'juste_leblanc',
    };
    const tree = createComponentWithIntl(<Avatar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
