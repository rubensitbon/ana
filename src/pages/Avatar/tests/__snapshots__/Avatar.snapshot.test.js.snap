// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page component should render correctly 1`] = `
<div
  className="sc-bdVaJa hiBQnU"
>
  <button
    onClick={[Function]}
    tabIndex={0}
  >
    <span>
      Retour
    </span>
  </button>
  <p />
  <p>
    <input
      className="github-avatar-input"
      onChange={[Function]}
      placeholder={undefined}
      type="text"
    />
  </p>
  <p>
    <button
      onClick={[Function]}
    >
      <span>
        Affiche l'avatar github
      </span>
    </button>
  </p>
  <img
    alt="user avatar"
    src="https://avatars1.githubusercontent.com/u/9215726?v=4"
  />
</div>
`;
